### ISSUE TITLE
Milestones and priority

### ISSUE THREAD

**Issue by inf3rno** [inf3rno/e2e-client#19]
In the #17 we discussed that bigger dependencies, for example module dependencies should be handled by milestones and smaller dependencies should be simply sub-issues. Only unexpected dependencies should be flagged with `status: blocked - dependency`. 

By priority we have 5 labels:

- priority 1: critical - serious bug or threat, must be dealt with asap
- priority 2: important - next issue to solve, it is blocking other issues
- priority 3: usual  - an average issue
- priority 4: delayed - not important now, but later it will be needed
- priority 5: optional - maybe later or maybe never

How to use these labels?

 - In a milestone we have a loose order for the tasks. For example we have a, b, c tasks. They can be more or less dependent on each other, or some of them can be more important than the other. We decide to do them in the a, b, c order. If so, then we can use p2 for "a", p3 for "b" and p4 for "c". So we will know by looking at the issues what we need to deal with first.
 - A normal enhance or bug issue has p3 label, or if it is important, then it can have a p2 label. When a bug is blocking a feature issue, then it will get the p2 label and the blocked issue will get a status flag. The p1 can be used for issues that must be solved before continuing everything else. For example in this case we need to discuss the workflow before starting to work on the project.
 - If we have 2 complex issues for modules which depend on each other a -> b, then we should solve this with the workflow instead of status flags. So we need to work on "a" before "b". In this case "b" should be in the next milestone. But what if the 2 modules don't work without each other or we want to release both at once. I think if so, then we need to use the status flag after all. We can use for example blocked on the tracking issues `a -> b` or if we need partial implementation of b or just a simple feature, then `a -> b3` and/or `a1 ->b2`. If `a1` is blocked in this case, then we don't have to add the flag to the tracking issue. If multiple sub-issues are blocked: `[a1,a3,a6] -> b2` then it is better to add the flag to the tracking issue and write about  the blocks there. Hmm I think in every case it might be better to write into the tracking issue, because if every module has it's own developer group, then nobody will check the sub-issues of another group. But if the main tracking issue of the other group is blocked by other modules, then that is interesting... Hmm so I think we can have a rule here. If a tracking issue of a big module is blocked, then that can be blocked only by a dependency, which is outside of the module. So for example another module or missing language feature, or something like that. If we have a bug in the module, which blocks a sub-issue, then only the sub-issue is blocked, but not the whole tracking issue. So if we have to implement multiple interdependent modules in a single milestone, then we can add status flags to the tracking issues when the modules block each other. Otherwise we use these blocking flags only on the sub-issues, for example if a bug blocks a sub-issue. The blocking flag is always added to a sub-issue, and after further investigation it can be added to the tracking issue too, if it turns out that it is caused by a module developed by somebody else. If so, we can add a bug issue, which is not a sub-issue of the blocked tracking issue and it must have p1 critical priority. If the blocking issue is not blocking a whole tracking issue, just a sub-issue, then the priority is just p2 important, or if the sub-issue has low priority, then p3 usual. So I think this is ok. We just need to document it in #18
